packages/react-instantsearch-dom/src/lib/infiniteHitsCache/sessionStorage.js

```diff
@@ -0,0 +1,49 @@
+import isEqual from 'react-fast-compare';
+
+function getStateWithoutPage(state) {
+  const { page, ...rest } = state;
+  return rest;
+}
+
+const KEY = 'ais.infinitehits';
+
+function hasSessionStorage() {
+  return (
+    typeof window !== 'undefined' &&
+    typeof window.sessionStorage !== 'undefined'
+  );
+}
+
+export default function getInfiniteHitsSessionStorageCache() {
+  return {
+    read({ state }) {
+      if (!hasSessionStorage()) {
+        return null;
+      }
+      try {
+        const cache = JSON.parse(window.sessionStorage.getItem(KEY));
+        return isEqual(cache.state, getStateWithoutPage(state))
+          ? cache.hits
+          : null;
+      } catch (_error) {
+        return null;
+      }
+    },
+    write({ state, hits }) {
+      if (!hasSessionStorage()) {
+        return;
+      }
+      try {
+        window.sessionStorage.setItem(
+          KEY,
+          JSON.stringify({
+            state: getStateWithoutPage(state),
+            hits,
+          })
+        );
+      } catch (error) {
+        console.error(error); // eslint-disable-line no-console
```

Comment:
Logging the error in the console is not useful, the user can't do anything with it like catching it or handling it differently. He will just get an error message without any possibility to fix/suppress it.

Since in that case the behavior is like having no cache I'm more to do nothing here (no `console.error`).

Reply:
It totally makes sense. Thanks for the feedback!
https://github.com/algolia/react-instantsearch/pull/2921/commits/fcd9bea0c31dd4f12fa7e62b9491732a91238349